File: opencv-learning/cpp/feature/blob.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;
int main()
{

    // Read image
    Mat im = imread( "D:/Develop/DL/deeplearning-notes/temp/images/blox.jpg", IMREAD_GRAYSCALE );
    // Set up the detector with default parameters.
//    SimpleBlobDetector detector;
//    // Detect blobs.
    std::vector<KeyPoint> keypoints;
//    detector.detect( im, keypoints);
    cv::Ptr<cv::SimpleBlobDetector> detector = cv::SimpleBlobDetector::create();
    detector->detect( im, keypoints );
    // Draw detected blobs as red circles.
    // DrawMatchesFlags::DRAW_RICH_KEYPOINTS flag ensures the size of the circle corresponds to the size of blob
    Mat im_with_keypoints;
    drawKeypoints( im, keypoints, im_with_keypoints, Scalar(0,0,255), DrawMatchesFlags::DRAW_RICH_KEYPOINTS );
    // Show blobs
    imshow("keypoints", im_with_keypoints );
    waitKey(0);
}

File: opencv-learning/feature/blob.py
# -*- coding:utf-8 -*-

import cv2
import numpy as np
detector = cv2.SimpleBlobDetector.create()
#keypoints = []

img = cv2.imread("temp/screen.png",cv2.IMREAD_GRAYSCALE)

keypoints = detector.detect(img)

imout = cv2.drawKeypoints(img, keypoints, np.array([]), (0,0,255), cv2.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS)

cv2.imshow("IMG",img)
cv2.imshow("OUT",imout)

cv2.waitKey()